### PR TITLE
Create sku name for publishing GPU VHD to SIG

### PR DESCRIPTION
**Pull request by bowang-666**
This is a hack to unblock me publishing GPU VHDs to SIG.
Long term wise consider move this logic out of yaml file, and have a group of build_mode, then set variables based on build_mode.

### CODE REVIEW

File: .pipelines/.vsts-vhd-builder.yaml [hunk 1]
```diff
@@ -138,7 +138,8 @@ phases:
           SKU_NAME="1804gen2"; else \
           OS_DISK_SAS="$(cat packer-output | grep "OSDiskUriReadOnlySas:" | cut -d " " -f 2)"; \
           VHD_NAME="$(echo $OS_DISK_SAS | cut -d "/" -f 8 | cut -d "?" -f 1)"; \
-          SKU_NAME=$UBUNTU_SKU; fi &&
+          SKU_NAME=$UBUNTU_SKU; fi && \
```
Comment (xuto2) [inline .pipelines/.vsts-vhd-builder.yaml:141]: extra "fi'?

Comment (bowang-666) [reply .pipelines/.vsts-vhd-builder.yaml:141]: this is the "fi" for previous if

File: .pipelines/.vsts-vhd-builder.yaml [hunk 2]
```diff
@@ -138,7 +138,8 @@ phases:
...
+          if [[ "${AZURE_VM_SIZE}" == *"Standard_NC"* ]]; then SKU_NAME="${SKU_NAME}gpu"; fi && \
           docker run --rm \
```
Comment (xuto2) [inline .pipelines/.vsts-vhd-builder.yaml:143]: i don't get it, what is this SKU_NAME for?

Comment (xuto2) [reply .pipelines/.vsts-vhd-builder.yaml:143]: ok, so this becomes SIG definition name. It seems we should use the "fullgpu" feature flags to tell if it's for GPU or not.
Also we need to have gen1 or gen2 in the sku name

Comment (bowang-666) [reply .pipelines/.vsts-vhd-builder.yaml:143]: update to use feature flag. gen2 was added at L138

**Pull request merged**